tools/scoring/score/dimensions/dimension.py

```diff
@@ -176,8 +251,107 @@ def is_entity_virtual(*, entity: EntityInstance) -> bool:
     """
     return entity.links is not None
 
+  @staticmethod
+  def match_virtual_entities(
+      *, solution_points_virtual: PointsVirtualList,
+      proposed_points_virtual: PointsVirtualList,
+      sort_candidates_by_key: str) -> Dict[float, List[_VirtualEntityMatch]]:
+    """
+    Finds the closest correlating virtual entities between two files
+    by comparing the intersections of raw field names contained therein.
+
+    Args:
+      solution_points_virtual: Raw field names and entity types
+        for all virtual entities in a file
+      proposed_points_virtual: Raw field names and entity types
+        for all virtual entities in a file
+      sort_candidates_by_key: Parameter by which to "break a tie"
+        if there are multiple matches with the same subscore.
+
+    Returns:
+      Dictionary whose keys are floats representing the extent to which the
+      provided entities correlated and whose values are lists of
+      _VirtualEntityMatch instances containing the metrics
+      by which those floats were calculated.
+    """
+    # Final pairings
+    matches_virtual = {None: []}
+
+    for solution_parameters in solution_points_virtual:
+      solution_raw_field_names, solution_entity_type = solution_parameters
+
+      # Record of entities evaluated and associated metrics
+      candidates = defaultdict(list)
+      # Running tally of which entity most closely corellates;
+      # even zero overlap will best the default at -1.0
+      best: float = -1.1
+
+      for proposed_parameters in proposed_points_virtual:
+        proposed_raw_field_names, proposed_entity_type = proposed_parameters
+
+        # Offload the quantification of intersection/difference
+        # and resulting subscore
+        subscore = _FieldsSubscore(
+            proposed_raw_field_names=proposed_raw_field_names,
+            solution_raw_field_names=solution_raw_field_names)
+
+        # Meanwhile, calculate a similar metric for overlap of types
+        types_score = _TypesSubscore(proposed_entity_type=proposed_entity_type,
+                                     solution_entity_type=solution_entity_type)
+
+        # If the current iteration matches better than the last,
+        # take note of its score for later lookup
+        if subscore.tally != 0 and subscore.tally > best:
+          best = subscore.tally
+
+        subscore_reference = _VirtualEntityMatch(
+            correct=subscore.correct,
```

Comment:
> now you can simplify this by providing only the following parameters:
> 
> ```
> _VirtualEntityMatch(subscore, types_score, solution_parameters, proposed_parameters)
> ```

Technically all I need is `correct_ceiling` and `types_score` because they are used to break ties when sorting — this is almost entirely for debugging.

> you can use a cls method ?
> 
> ```
> @classmethod
> def FromScores(subscore, types_score, solution_parameters, proposed_parameters)
> ....
> ```

I don't think I follow.